src/main/java/org/wildfly/security/auth/realm/ldap/LdapSecurityRealm.java

```diff
@@ -124,17 +125,16 @@ public ModifiableRealmIdentity getRealmIdentityForUpdate(String name, final Prin
     }
 
     @Override
-    public Iterator<ModifiableRealmIdentity> getRealmIdentityIterator() throws RealmUnavailableException {
+    public CloseableIterator<ModifiableRealmIdentity> getRealmIdentityIterator() throws RealmUnavailableException {
         if (identityMapping.iteratorFilter == null) {
             throw log.ldapRealmNotConfiguredToSupportIteratingOverIdentities();
         }
 
-        return new Iterator<ModifiableRealmIdentity>() {
+        return new CloseableIterator<ModifiableRealmIdentity>() {
```

Comment:
Yes, current implementation of LdapRealm iterating doesnt need to be closed. It is actually required only for FileSystemRealm now. (But I think it will be better if higher layers (subsystem) can just call close() on any iterable realm.)